modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/index/contributor/AuditedModelDocumentContributor.java

```diff
@@ -35,6 +41,8 @@ public void contribute(
 		document.addKeyword(Field.COMPANY_ID, auditedModel.getCompanyId());
 		document.addDate(Field.CREATE_DATE, auditedModel.getCreateDate());
 		document.addDate(Field.MODIFIED_DATE, auditedModel.getModifiedDate());
+		document.addKeyword(
+			"userExternalReferenceCode", _getUserERC(auditedModel));
```

Comment:
I am worried about this database requisition to get external reference codes, it doesn't make sense to hit the database once we have the `auditedModel`. Could we add a `getUserExternalReferenceCode` method to the interface and implement it? would it be to much work?

Reply:
\cc @BryanEngler

Reply:
So adding getUserExternalReferenceCode to the AuditedModel would mean changing 896 classes. Additionally, that alone wouldn't help avoid a database hit as these implementations would still likely be calling the userLocalService (see the [DLFolderModelImpl.getUserUuid](https://github.com/liferay/liferay-portal/blob/4cb0a9753eacfd39efc7f717d22c2bcefb0c7f82/portal-impl/src/com/liferay/portlet/documentlibrary/model/impl/DLFolderModelImpl.java#L656-L665)). To remove the database hit we would need to store the userUuid into each to these models.

Keep in mind that we do cache this information.

Reply:
Thanks for the explanation, I didt  know it was cached. If this is the case, it should be just fine and we wouldn't need to worry about additional database hits. Thx!